### PR TITLE
Add automatic PR labelling action

### PR DESCRIPTION
**Pull request by teodutu**
Uses [`labeler`](https://github.com/actions/labeler) for assigning `topic/*` and `area/*` labels and a [custom script](https://github.com/actions/github-script) for `kind/*` labels.
`kind/new` is assigned to any PR that contains a newly added file and `kind/improvement` is assigned to any PR that contains at least one file only with additions. The `student-contrib` and `kind/fix` labels are to be added manually.

Fixes https://github.com/open-education-hub/operating-systems/issues/89.

Signed-off-by: Teodor Dutu <teodor.dutu@gmail.com>

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: "Pull Request Labeler"
+on: [pull_request]
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:2]: As mentioned [here] this method won't work with `pull_request`. You should use something like [this] to schedule the workflow to run from time to time. Also I would add a workflow_dispatch just for testing purposes to check that this is working before merging it.

[here]: https://github.com/actions/labeler/issues/12
[this]: https://github.com/actions/starter-workflows/pull/78/files

**Comment by teodutu** [comment]
I tested the scripts with a PR in my own fork, where the `GITHUB_TOKEN` works: https://github.com/teodutu/operating-systems/pull/6.

**Review by gabrielmocanu — APPROVED** [review]
I think we can integrate this.

**Comment by gabrielmocanu** [comment]
We need this one faster

**Pull request merged**